components/apimgt/org.wso2.carbon.apimgt.solace/src/main/java/org/wso2/carbon/apimgt/solace/SolaceAdminApis.java

```diff
@@ -0,0 +1,824 @@
+        } catch (IOException | APIManagementException e) {
+            log.error(e.getMessage());
+        }
+        return null;
+    }
+
+    /**
+     * Register API in Solace using AsyncAPI Definition
+     *
+     * @param organization  name of the Organization
+     * @param title         name of the Solace API
+     * @param apiDefinition Async definition of the Solace API
+     * @return CloseableHttpResponse of the PUT call
+     */
+    public CloseableHttpResponse registerAPI(String organization, String title, String apiDefinition) {
+        URL serviceEndpointURL = new URL(baseUrl);
+        HttpClient httpClient = APIUtil.getHttpClient(serviceEndpointURL.getPort(), serviceEndpointURL.getProtocol());
+        HttpPut httpPut = new HttpPut(baseUrl + "/" + organization + "/apis/" + title);
+        httpPut.setHeader(HttpHeaders.AUTHORIZATION, "Basic " + getEncoding());
+        httpPut.setHeader(HttpHeaders.CONTENT_TYPE, "text/plain");
+        JsonNode jsonNodeTree;
+        String jsonAsYaml = null;
+        // convert json to yaml
+        try {
+            jsonNodeTree = new ObjectMapper().readTree(apiDefinition);
+            jsonAsYaml = new YAMLMapper().writeValueAsString(jsonNodeTree);
+        } catch (JsonProcessingException e) {
+            log.error(e.getMessage());
+        }
+        //add definition to request body
+        if (jsonAsYaml != null) {
+            StringEntity params = null;
+            try {
+                params = new StringEntity(jsonAsYaml);
+            } catch (UnsupportedEncodingException e) {
+                log.error(e.getMessage());
+            }
+            httpPut.setEntity(params);
+            try {
+                return APIUtil.executeHTTPRequest(httpPut, httpClient);
+            } catch (IOException | APIManagementException e) {
+                log.error(e.getMessage());
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Create API Product in Solace using AsyncAPI Definition
+     *
+     * @param organization           name of the Organization
+     * @param environment            name of the Environment
+     * @param apiProductName         name of the API product
+     * @param apiNameForRegistration name of the Solace API product
+     * @param aai20Document          Async definition of the Solace API
+     * @return CloseableHttpResponse of the POST call
+     */
+    public CloseableHttpResponse createAPIProduct(String organization, String environment, Aai20Document aai20Document,
+                                         String apiProductName, String apiNameForRegistration) {
+        URL serviceEndpointURL = new URL(baseUrl);
+        HttpClient httpClient = APIUtil.getHttpClient(serviceEndpointURL.getPort(), serviceEndpointURL.getProtocol());
+        HttpPost httpPost = new HttpPost(baseUrl + "/" + organization + "/apiProducts");
+        httpPost.setHeader(HttpHeaders.AUTHORIZATION, "Basic " + getEncoding());
```

Comment:
is this correct getEncoding()

Reply:
@tharindu1st Yes. This is a newly written private method that is used for avoiding repetition in HTTP request creation. Not the base64.encoding method.